sympy/assumptions/sathandlers.py

```diff
@@ -300,6 +300,9 @@ def register_fact(klass, fact, registry=fact_registry):
     (Mul, Equivalent(Q.zero, AnyArgs(Q.zero))),
     (MatMul, Implies(AllArgs(Q.square), Equivalent(Q.invertible, AllArgs(Q.invertible)))),
     (Add, Implies(AllArgs(Q.positive), Q.positive)),
+    (Add, Implies(AllArgs(Q.positive), ~Q.negative)),
+    (Add, Implies(AllArgs(Q.negative), Q.negative)),
+    (Add, Implies(AllArgs(Q.negative), ~Q.positive)),
```

Comment:
This fact also ought to be redundant. The assumptions should already know that negative >> ~positive.